lib/quicly.c

```diff
@@ -611,18 +613,25 @@ size_t quicly_decode_packet(quicly_context_t *ctx, quicly_decoded_packet_t *pack
             goto Error;
         packet->cid.src.base = (uint8_t *)src;
         src += packet->cid.src.len;
-        if (ctx->cid_encryptor != NULL) {
-            ctx->cid_encryptor->decrypt_cid(ctx->cid_encryptor, &packet->cid.dest.plaintext, packet->cid.dest.encrypted.base,
-                                            packet->cid.dest.encrypted.len);
-        } else {
-            packet->cid.dest.plaintext = (quicly_cid_plaintext_t){0};
-        }
         switch (packet->octets.base[0] & QUICLY_PACKET_TYPE_BITMASK) {
         case QUICLY_PACKET_TYPE_INITIAL:
         case QUICLY_PACKET_TYPE_0RTT:
+            if (ctx->cid_encryptor != NULL && packet->cid.dest.encrypted.len != 0 &&
```

Comment:
What if `ctx->cid_encryptor == NULL` or `packet->cid.dest.encrypted.len == 0`? Don't we want to set `quicly_cid_plaintext_invalid` as well?

Reply:
Thank you for spotting that! Fixed in ee33316.